app/models/log.rb

```diff
@@ -44,17 +43,8 @@ class Log < ApplicationRecord
 
   belongs_to :user
 
-  has_many :number_log_entries,
-    class_name: 'LogEntries::NumberLogEntry',
-    dependent: :destroy,
-    inverse_of: :log
-
-  has_many :text_log_entries,
-    class_name: 'LogEntries::TextLogEntry',
-    dependent: :destroy,
-    inverse_of: :log
-
-  has_many :log_shares, dependent: :destroy
+  has_many :log_entries, dependent: :destroy, inverse_of: :log
```

Comment:
I think that explicitly specifying this `inverse_of` option is essential for something to work correctly.